### ISSUE TITLE
FR: How to disable/override keyboard navigation on `igc-dockmanager`?

### ISSUE THREAD

**Issue by luiscla27** [IgniteUI/igniteui-dockmanager#99]
Hi!

This a Feature Request to make the[ KEYBOARD NAVIGATION of DockManager](https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/layouts/dock-manager#docking) configurable. Ideally, the shortcuts used should be configurable by the developer implementing the component, and even disable them! like instead of using `ctrl + shift + arrow left`, maybe that shortcut is already occupied, and I want to use another one.

I'm using the Angular implementation of Dock Manager, also read the web-component docs. But, I couldn't find a way to turn off the keyboard navigation:
- https://www.infragistics.com/products/ignite-ui-angular/angular/components/dock-manager
- https://www.infragistics.com/products/ignite-ui-web-components/web-components/components/layouts/dock-manager#docking

I want to disable all of it.

The reason is that I'm using it to put text boxes inside of it, and `ctrl + shift + arrow left/right` is also a shortcut used by certain users to move the cursor to words around the textbox. This issue had happened in other products, and usually occurs due to bugs, this are some examples:
- Same [issue on Notepad++ ](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/4649) (fixed).
- Same [issue on VSCode](https://github.com/microsoft/vscode/issues/41199) (fixed).

The reason I'm reporting this as a FR, instead of a BUG, is because that it seems to me that this behaviour was intended, as maybe you didn't acknowledge my use case to exist at all. Being that, to put text boxes inside the dockable panels.

Please let me know if I should move this [the Angular repo](https://github.com/IgniteUI/igniteui-angular/issues).

- [ ] Research
- [x] Specification
- [ ] Component and tests implementation
- [x] API Documentation
- [x] Design implementation

**Comment by gedinakova** [comment]
Hello @luiscla27,
Thank you for the detailed report of the issue you've encountered.

Currently the DockManager exposes the ["disableKeyboardNavigation" property](https://www.infragistics.com/products/ignite-ui/dock-manager/docs/typescript/latest/classes/IgcDockManagerComponent.html#disableKeyboardNavigation) which can be used to disable keyboard navigation as a whole as shown in [this sample application.](https://stackblitz.com/edit/fohtk3?file=src%2Fapp%2Fdock-manager-sample%2Fdock-manager.component.html)

If disabling all shortcuts is not exactly the desired behavior, for more detailed handling of keyboard interactions I would suggest you handle the keydown event and fine-tune the behavior to match your application's needs.

For more details, please refer to [this issue](https://github.com/IgniteUI/igniteui-dockmanager/issues/22) which actually led to adding the "disableKeyboardNavigation" property.

Should you need any further assistance, please, do not hesitate to let us know!
Thanks in advance.
